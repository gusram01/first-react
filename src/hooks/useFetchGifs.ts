import { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';
import { RespDataGiphy } from '../types/auxTypes';

interface IGiphy {
  data: Partial<RespDataGiphy>[];
  loading: boolean;
}

export const useFetchGifs = (category: string) => {
  const [State, setState] = useState<IGiphy>({
    data: [],
    loading: true
  });

  useEffect(() => {
    getGifs(category)
      .then(img => {
        setState({ data: img, loading: false })
      })
  }, [category]);

  return State;
}