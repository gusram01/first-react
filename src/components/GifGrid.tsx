import React from 'react';
import GifGridItem from './GiFGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { RespDataGiphy } from '../types/auxTypes';
import PropTypes from 'prop-types';

interface Props {
  category: string;
}

const GifGrid: React.FC<Props> = ({ category }) => {

  const { data: images, loading } = useFetchGifs(category);
  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {loading && <p className="animate__animated animate__flash">Loading...</p>}
      <div className="card-grid">

        {
          images.map((image: Partial<RespDataGiphy>) =>
            < GifGridItem
              key={image.id}
              img={image} />
          )
        }
      </div>
    </>
  )

}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}

export default GifGrid;