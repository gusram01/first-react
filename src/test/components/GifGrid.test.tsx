import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('GifGrid tests', () => {
  const category = 'something';
  let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

  test('should render GifGrid correctly', () => {
    //@ts-expect-error
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render items when images is loaded from UseFetchGifs', () => {
    //@ts-expect-error
    useFetchGifs.mockReturnValue({
      data: [
        {
          id: 'ABC',
          url: 'https://somepage.com/some.jpg',
          title: 'some title'
        }
      ],
      loading: false
    });
    wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(1);
  });
});