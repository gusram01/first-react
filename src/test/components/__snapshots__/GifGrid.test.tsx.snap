// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid tests should render GifGrid correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    something
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`GifGrid tests should render items when images is loaded from UseFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    something
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      img={
        Object {
          "id": "ABC",
          "title": "some title",
          "url": "https://somepage.com/some.jpg",
        }
      }
      key="ABC"
    />
  </div>
</Fragment>
`;
